apps/onlyoffice.sh

```diff
@@ -0,0 +1,294 @@
+then
+    echo "aufs" >> /etc/modules
+fi
+
+# Set docker storage driver to AUFS
+AUFS2=$(grep -r "aufs" /etc/default/docker)
+if ! [ "$AUFS2" = 'DOCKER_OPTS="--storage-driver=aufs"' ]
+then
+    echo 'DOCKER_OPTS="--storage-driver=aufs"' >> /etc/default/docker
+    service docker restart
+fi
+
+# Check of docker runs and kill it
+DOCKERPS=$(docker ps -a -q)
+if [ "$DOCKERPS" != "" ]
+then
+    echo "Removing old Docker instance(s)... ($DOCKERPS)"
+    any_key "Press any key to continue. Press CTRL+C to abort"
+    docker stop "$DOCKERPS"
+    docker rm "$DOCKERPS"
+fi
+
+# Disable Onlyoffice if activated
+if [ -d "$NCPATH"/apps/onlyoffice ]
+then
+    sudo -u www-data php "$NCPATH"/occ app:disable onlyoffice
+    rm -r "$NCPATH"/apps/onlyoffice
+fi
+
+# Install Onlyoffice docker
+docker pull onlyoffice/documentserver
+docker run -i -t -d -p 127.0.0.3:9090:80 -p 127.0.0.3:9091:443 onlyoffice/documentserver
+
+# Install Apache2
+if [ "$(dpkg-query -W -f='${Status}' apache2 2>/dev/null | grep -c "ok installed")" == "1" ]
+then
+    sleep 0.1
+else
+    {
+    i=1
+    while read -r line; do
+        ((i++))
+        echo $i
+    done < <(apt install apache2 -y)
+    } | whiptail --title "Progress" --gauge "Please wait while installing Apache2" 6 60 0
+fi
+
+# Enable Apache2 module's
+a2enmod proxy
+a2enmod proxy_wstunnel
+a2enmod proxy_http
+a2enmod ssl
+
+# Create Vhost for OnlyOffice online in Apache2
+if [ ! -f "$HTTPS_CONF" ];
+then
+    cat << HTTPS_CREATE > "$HTTPS_CONF"
+<VirtualHost *:443>
+     ServerName $SUBDOMAIN:443
+
+     SSLEngine on
+     ServerSignature On
+     SSLHonorCipherOrder on
+
+  SSLCertificateChainFile $CERTFILES/$SUBDOMAIN/chain.pem
+  SSLCertificateFile $CERTFILES/$SUBDOMAIN/cert.pem
+  SSLCertificateKeyFile $CERTFILES/$SUBDOMAIN/privkey.pem
+  SSLOpenSSLConfCmd DHParameters $DHPARAMS
+  SSLProtocol             all -SSLv2 -SSLv3
+  SSLCipherSuite ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS
+
+     LogLevel warn
+     CustomLog ${APACHE_LOG_DIR}/access.log combined
+     ErrorLog ${APACHE_LOG_DIR}/error.log
+
+# Just in case - see below
+SSLProxyEngine On
+SSLProxyVerify None
+SSLProxyCheckPeerCN Off
+SSLProxyCheckPeerName Off
+
+# contra mixed content warnings
+RequestHeader set X-Forwarded-Proto "https"
+
+# basic proxy settings
+ProxyRequests off
+
+        ProxyPass / http://127.0.0.3:9090/
```

Comment:
You use http and wss (websocket secure like https but then for sockets), you you want this?